src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/PartyAddons.kt

```diff
@@ -0,0 +1,228 @@
+                    PartyMember(
+                        name,
+                        PartyMemberType.MODERATOR,
+                        status,
+                        rank
+                    )
+                )
+            }
+            event.isCanceled = true
+        } else if (message.startsWith("§eParty Members: ")) {
+            playerPattern.findAll(message.substringAfter("§eParty Members: ")).forEach {
+                val name = it.groups["name"]?.value ?: return
+                val status = it.groups["status"]?.value ?: return
+                val rank = it.groups["rank"]?.value ?: return
+
+                party.add(
+                    PartyMember(
+                        name,
+                        PartyMemberType.MEMBER,
+                        status,
+                        rank
+                    )
+                )
+            }
+            event.isCanceled = true
+        } else if (message.startsWith("§cYou are not currently in a party.") && awaitingMessage != 0) {
+            party.clear()
+        } else if (message == "§9§m-----------------------------------------------------§r" && awaitingMessage != 0) {
+            awaitingMessage--
+            if (awaitingMessage == 1 || party.isEmpty()) return
+
+            val component = UMessage("§aParty members (${party.size})\n")
+
+            val player = party.first { it.name == Skytils.mc.thePlayer.name }
+
+            if (player.type == PartyMemberType.LEADER) {
+                component.append(
+                    createButton(
+                        "§9[Party Warp] ",
+                        "/p warp",
+                        "§9Click to warp the party."
+                    )
+                ).append(
+                    createButton(
+                        "§e[Toggle All Invite]\n",
+                        "/p settings allinvite",
+                        "§eClick to toggle all invite."
+                    )
+                ).append(
+                    createButton(
+                        "§6[Toggle Mute] ",
+                        "/p mute",
+                        "§6Click to toggle mute."
+                    )
+                ).append(
+                    createButton(
+                        "§c[Kick Offline] ",
+                        "/p kickoffline",
+                        "§cClick to kick offline members."
+                    )
+                ).append(
+                    createButton(
+                        "§4[Disband]\n",
+                        "/p disband",
+                        "§4Click to disband the party."
+                    )
+                )
+            }
+
+            val partyLeader = party.first { it.type == PartyMemberType.LEADER }
+            component.append(
+                "\n${partyLeader.status}➡§r ${partyLeader.rank}${partyLeader.name}"
+            )
+
+            if (party.any { it.type == PartyMemberType.MODERATOR }) component.append("\n§eMods")
```

Comment:
iterates through the list twice for no reason, maybe save the result of the filter